File: 242-valid-anagram/242-valid-anagram.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length()!=t.length()) return false;

        s.toLowerCase();
        t.toLowerCase();

        char[] tArray = t.toCharArray();
        char[] sArray = s.toCharArray();


        Arrays.sort(tArray);
        Arrays.sort(sArray);

        boolean result = Arrays.equals(tArray, sArray);
        
        return result;
        
    }
}

File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        map = {}
        
        for ch in s:
            map[ch] = 1 + map.get(ch, 0)
        
        for ch in t:
            if ch in map:
                if map[ch] > 0:
                    map[ch] -= 1
                else:
                    return False
            else:
                return False
        
        return True